cluster.go

```diff
@@ -942,7 +943,10 @@ func (c *Cluster) StateSync(ctx context.Context) error {
 		_, tracked := trackedPinsMap[pin.Cid.String()]
 		if !tracked {
 			logger.Debugf("StateSync: tracking %s, part of the shared state", pin.Cid)
-			c.tracker.Track(ctx, pin)
+			err = c.tracker.Track(ctx, pin)
+			if err == maptracker.ErrFullQueue {
```

Comment:
You should not import an specific pintracker here. The Cluster only knows about PinTracekrs, not their particular implementations. Moreover, since you don't know how a pintracker is implemented, you should abort on any error,